core/blockchain.go

```diff
@@ -2336,24 +2336,25 @@ func (bc *BlockChain) UpdateValidatorVotingPower(
 				return nil, err
 			}
 
-			aprComputed, err := apr.ComputeForValidator(
-				bc, block, wrapper,
-			)
-			if err == nil && aprComputed != nil {
-				a := *aprComputed
+			stats.APR = numeric.ZeroDec()
+			if wrapper.TotalDelegation().Cmp(common.Big0) > 0 {
```

Comment:
actually this TotalDelegation is an expensive for-loop. Really we should just check if wrapper.Delegations[0].Cmp() > 0 

then its constant time. Let's think careful if can use this assumption

Reply:
done.